content/ui-patterns/degraded-experiences.mdx

```diff
@@ -0,0 +1,535 @@
+  alt="Image of the notification link button in the global nav without a badge"
+  src="https://github.com/primer/design/assets/2313998/4f39547c-749b-4e44-a15f-05d18a1dc0c1"
+/>
+
+</Box>
+
+### Replacing UI with a message
+
+<img
+  width="960"
+  alt="A screenshot of the Primer organization page. The repositories list is replaced with a message informing the user that the repositories are temporarily unavailable."
+  src="https://github.com/primer/design/assets/2313998/3a2ce19a-2699-483a-a967-5b2e7e3d86d8"
+/>
+
+We don't want users to think they've suffered data loss. If we know a user created something, don't show a generic "empty state". It's better to explain that it's unavailable, or remove the entire section from the page (including the section's heading). For example, I might think my repositories were deleted if I'm on an organization page (like [https://github.com/primer/](https://github.com/primer/)) and the repositories aren't listed there.
+
+Ideally we can strike a balance and give the user just the right amount of context without overwhelming them with error messages. A page with too many error messages could communicate an unnecessarily reactionary and negative tone. As a general guideline, we suggest limiting pages to 5 or less outage messages.
+
+#### Inline content and smaller elements
+
+Smaller parts of the UI that cannot be accurately rendered but are too important to exclude entirely can often be replaced with a short error message.
+
+<img
+  width="960"
+  alt="Two images: 1. Image of Memex table with populated content cells; 2. Image of Memex table with missing content message in cells"
+  src="https://github.com/primer/design/assets/2313998/9c2feb5b-dbc0-46a8-b184-070bd90484fe"
+/>
+
+By default, replace the affected content with an error message. Show a warning icon before the message to help differentiate it from non-degraded content. The message may be colored with `fg.warning` to draw more attention to it.
+
+Be mindful that rendering too many error messages on the page in `fg.warning` could be jarring and make the page feel broken instead of degraded.
+
+<DoDontContainer>
+  <Do>
+    <img
+      src="https://github.com/primer/design/assets/2313998/c85a8bf4-cdf4-48de-9e18-b9547ab371e3"
+      alt="Image of Memex table with missing content message in cells"
+      width="456"
+    />
+    <Caption>Render an error message in place of the content.</Caption>
+  </Do>
+  <Dont>
+    <img
+      src="https://github.com/primer/design/assets/2313998/6a0cc580-3a0a-4023-9e34-e6387e0629b5"
+      alt="Image of Memex table with 'undefined of undefined'"
+      width="456"
+    />
+    <Caption>Don't attempt to render UI that is missing critical information.</Caption>
+  </Dont>
+</DoDontContainer>
+
+#### Panels and larger areas
+
+<img
+  width="960"
+  alt="Two images: 1. comment box; 2. blankslate in place of comment box"
+  src="https://github.com/primer/design/assets/2313998/8321cdb9-8bc2-4f7c-8107-ff535d12498f"
+/>
```

Comment:
Nitpick: Image 1 is a bit blurry and image 2 has a bit funny border spacing. Not blocking.

Reply:
I'm not seeing either of these issues on my end. Maybe we can attempt a screen-share, or you could send a marked up screenshot?